### PR TITLE
[1.17] Backport: Add container out of memory metrics

### PR DESCRIPTION
**Pull request by discordianfish**
Backport of #4690 to 1.17

---
This patch adds a new metric
`container_runtime_crio_containers_oom_total` and
`container_runtime_crio_containers_oom`, which collects out of memory
(oom) metrics global and by container name. This also includes
sandboxes, since we reuse the CRI-O internal name for every container.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

### CODE REVIEW

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -94,6 +100,55 @@ var (
...
+	// CRIOImageLayerReuse collects image pull metrics for every resused image layer
+	CRIOImageLayerReuse = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: subsystem,
+			Name:      CRIOImageLayerReuseKey,
+			Help:      "Reused (not pulled) local image layer count by name",
+		},
+		[]string{"name"},
+	)
+
```
Comment (haircommander) [inline server/metrics/metrics.go:133]: these all seem unused. am I seeing that right?

**Comment by haircommander** [comment]
context: https://kubernetes.slack.com/archives/CAZH62UR1/p1638357795128600

**Comment by saschagrunert** [comment]
A friendly reminder that this PR had no activity for 30 days.

**Pull request closed without merging**